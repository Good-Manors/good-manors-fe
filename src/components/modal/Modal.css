.modalOverlay {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1040;
  width: 100vw;
  height: 100vh;
  background-color: #000;
  opacity: .5;
}

.modalWrapper {
  position: fixed;
  top: 0;
  left: 0;
  z-index: 1050;
  width: 100%;
  height: 100%;
  overflow-x: hidden;
  overflow-y: auto;
  outline: 0;
}

.modal {
  z-index: 100;
  background: white;
  position: relative;
  margin: 30vh auto;
  border-radius: 3px;
  max-width: 700px;
  padding: 40px;
  border-bottom: 20px solid #8ab4b1;

  h3 {
    font-size: 1.6em;
  }

  input {
    width: 300px;
  }

  button {
    text-align: center;
  }

}

.modalHeader {
  display: flex;
  justify-content: flex-end;
}

.modalCloseButton {
  font-size: 1.4rem;
  font-weight: 700;
  line-height: 1;
  color: black;
  background-color: transparent;
  opacity: .6;
  border-radius: 50%;
  border: transparent;
  cursor: pointer;
  margin: 0;
}

.radio {
  display: flex;
  justify-content: space-evenly;
}

.form {
  text-align: center;
}

.column {
  text-align: left;

  input {
    width: 20px;
  }

  .textBox {
    width: 200px;
  }
}
