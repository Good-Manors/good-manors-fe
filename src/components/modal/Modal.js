import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import ReactDOM from 'react-dom';
import styles from './Modal.css';
import Form1 from './Form1';
import Form2 from './Form2';
import Form3 from './Form3';
import Form4 from './Form4';
import { initializeHome } from '../../services/homes';
import { setHome } from '../../actions/homeActions';

const Modal = ({ isShowing, hide }) => {
  const [currentStep, setCurrentStep] = useState(1);
  const [name, setName] = useState('');
  const [drawer, setDrawer] = useState([]);
  const [card, setCard] = useState([]);
  const [redirect, setRedirect] = useState(false);
  const dispatch = useDispatch();

  const handleChange = ({ target }) => {
    setName(target.value);
  };

  const handleForm = (step) => {
    if(step === 2) {
      const selectedDrawers = Array.from(document.querySelectorAll('input:checked')).map(drawer => {
        return drawer.value;
      });
      setDrawer(selectedDrawers);
    }
    if(step > 2) {
      const selectedCards = Array.from(document.querySelectorAll('input:checked')).map(card => {
        return card.value;
      });
      setCard([...card, selectedCards]);
    }
    setCurrentStep(step + 1);
  };

  const handleFormSubmit = () => {
    initializeHome(name, drawer, card)
      .then(home => {
        dispatch(setHome(home));
        hide();
        setRedirect(true);
      });
  };

  const drawerSteps = drawer.map((eachDrawer, i) => {
    return <Form3 index={i} key={i} drawerName={eachDrawer} currentStep={currentStep} card={card} handleForm={handleForm} />;
  });

  if(redirect) return <Redirect to="/home" />;

  return (
    isShowing ? ReactDOM.createPortal(
      <>
        <div className={styles.modalOverlay} />
        <div className={styles.modalWrapper} aria-modal aria-hidden tabIndex={-1} role="dialog">
          <div className={styles.modal}>

            <div className={styles.modalHeader}>
              <button type="button" className={styles.modalCloseButton} data-dismiss="modal" aria-label="Close" onClick={hide}>
                <span aria-hidden="true">&times;</span>
              </button>
            </div>

            <Form1 handleChange={handleChange} handleForm={handleForm} name={name} currentStep={currentStep} />
            <Form2 currentStep={currentStep} handleForm={handleForm} />
            {drawerSteps}
            <Form4 currentStep={currentStep} drawer={drawer} card={card} handleFormSubmit={handleFormSubmit} />
          </div>
        </div>
      </>, document.body
    ) : null
  );
};

export default Modal;
