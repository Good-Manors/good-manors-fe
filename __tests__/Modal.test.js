import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import Modal from '../src/components/modal/Modal';

describe('Modals', () => {
  it('Modal.js', () => {
    jest.mock('react-redux', () => ({
      useDispatch: ()=> { },
      useSelector: () => ({})
    }));
    const getWrapper = (mockStore = createStore()) => mount(
      <Provider store={mockStore}>
        <Modal />
      </Provider>
    );
    expect(getWrapper).toMatchSnapshot();
  });


  
});
