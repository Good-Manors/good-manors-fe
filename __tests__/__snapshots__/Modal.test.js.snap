// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modals Modal.js 1`] = `[Function]`;
